Update shelly.markdown with clearer language and more actionable steps

chefdeit opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Improve shelly.markdown with actionable troubleshooting steps for gen 1 and gen 2 Shelly devices.



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the Brands repository:
- This PR fixes or closes issue: further helps Issue #97887

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

chefdeit committed:
Update shelly.markdown steps per @thecode advice

Per https://github.com/home-assistant/core/issues/97887#issuecomment-1667295820 rolled the slightly clearer steps into the doc

chefdeit committed:
Update shelly.markdown with clearer language and further troubleshooting steps

Add more specific & actionable UDP propagation and Wi-Fi troubleshooting ideas.

chefdeit committed:
Merge pull request #1 from Chef-de-IT/Chef-de-IT-patch-1

Update shelly.markdown steps per @thecode advice

chefdeit committed:
Merge branch 'home-assistant:current' into current

Chef-de-IT commented:
> It seems that this PR is targeted against an incorrect branch since it has a parent PR on one of our codebases. Documentation that needs to be updated for an upcoming release should target the `next` branch. Please change the target branch of this PR to `next` and rebase if needed.

This documentation update is for the current release.

bieniu commented:
You need to fix lint errors:
![obraz](https://github.com/home-assistant/home-assistant.io/assets/478555/cb985ee5-3482-4356-b9eb-fd1ba6e582c7)

chefdeit committed:
Fix ordered list markers in shelly.markdown

Change from
1. list item
1. list item
1. list item
To
1. list item
2. list item
3. list item
... because lint complained, even though it displays fine

thecode commented:
Link to preview of the updated docs: https://deploy-preview-28487--home-assistant-docs.netlify.app/integrations/shelly/

thecode commented:
Looking at the docs I think we are getting too much complicated now for most users, we should think how to reorganize this section, maybe put all the debug instructions as a last chapter

chefdeit committed:
Update shelly.markdown move troubleshooting steps to the bottom

Troubleshooting list is moved to the bottom of the doc as per the last comment

Chef-de-IT commented:
> Looking at the docs I think we are getting too much complicated now for most users, we should think how to reorganize this section, maybe put all the debug instructions as a last chapter

@thecode done

Chef-de-IT commented:
I might have removed self as the Assignee - wasn't intentional and if needed, feel free to add me back in. Sorry & thanks

Chef-de-IT commented:
> I think we need to split this into two parts, one describing the configuration and the second describing how to debug/fix the problem with CoIoT, and put it in a separate chapter/section.

@bieniu as of the last update, the debug and troubleshooting section is in a separate chapter/section at the bottom of the doc. With that said, I too want to see HA troubleshooting docs getting better to infinity - namely, I'd love to see all troubleshooting documentation and steps across HA to be standardized & adhering to a common pattern and presentation format of:

- [ ] Symptoms →
- [ ] Triage Steps →
- [ ] Root Cause →
- [ ] Solution Options →
- [ ] Steps for a solution option →

However, in the spirit of https://developers.home-assistant.io/docs/review-process/ and given there are users having this problem who could use the already enhanced doc yesterday, how about we start with this PR getting to the users first ASAP, and then we can continue to evolve it?

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,30 +43,43 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
 
-## Shelly device configuration (generation 1)
+- In Home Assistant UI, on the Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`;
+- 2nd generation devices will have a web UI sidebar on the left;
+- 2nd generation devices will have a blue Shelly favicon on their web browser tab.
```
In my opinion, a better way to distinguish the generation of devices is the model name, if the name includes Plus/Pro, it is a Gen2 device. Users probably know what device they bought, they should know that IMO.

thecode replied:
We should not point to web colors, @Chef-de-IT current Shelly beta which is starting role out already changed the web colors from previous version due to major web re-design. The HA gen1/gen2 or Plus/Pro in the name is the correct way.

chefdeit replied:
> if the name includes Plus/Pro, it is a Gen2 device

[bieniu](https://github.com/bieniu) , in the live version of this doc last section https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/shelly.markdown#known-issues-and-limitations I'm reading:

> Generation 1 "Shelly 4Pro" and "Shelly Sense" are not supported (devices based on old CoAP v1 protocol)

I may be wrongly conflating "old protocol" with "it makes no sense for this to be the latter generation device". Is Shelly 4Pro such as it's mentioned in the quoted doc, generation 2, generation 1, or pre-generation 1 (say generation 0)?? 

Shelly doesn't print generation 1 vs 2 on the product carton or even among the specs they list on the website - to say nothing of the listing on Amazon. As I type this, I'm holding a _**Shelly** dimmer **2**_ box (Shelly and 2 are imprinted in bold), yet it's generation 1. Given all of this, how can we realistically say "user should know" ? Know how?

I feel, the point of a doc, especially a troubleshooting doc, is to help as broad a range of users in trouble as possible, to resolve their issues and go on having a good HA experience. The rest of us nerds can browse the code.

chefdeit committed:
Update shelly.markdown with small clarifications

Switched to native GitHub alert box syntax for notes and warnings
Reordered a note re no cloud needed to a less confusing place higher up

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,45 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+- 2nd generation devices will have a blue Shelly favicon on their web browser tab.
 
-Generation 1 devices use the `CoIoT` protocol to communicate with the integration. For Shelly firmware 1.10.0 or newer, `CoIoT` must be enabled in the device settings. Navigate to the local IP address of your Shelly device, **Internet & Security** >> **ADVANCED - DEVELOPER SETTINGS** and check the box **Enable CoIoT**.
+> [!NOTE]  
```
You should use `<div class="note"></div>` intead of `[!NOTE]`.

chefdeit replied:
I'll make these changes throughout the doc. Was initially under the impression that relying on GitHub's native alert box markdown ( https://github.com/orgs/community/discussions/16925 ) instead of styled divs was semantically cleaner, but will switch back to divs.

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,45 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+6. Click **Apply** to save the settings;
+7. Reboot the Shelly device.
+
+> [!NOTE]  
```
You should use `<div class="note"></div>` instead of `[!NOTE]`.

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +312,36 @@ Please check from the device Web UI that the configured server is reachable.
+> [!WARNING]  
+> Leaving the Debug Log enabled for an extended period will age the Shelly device's flash memory, i.e. use up its finite write endurance cycles.
+
+> [!NOTE]  
```
You should use `<div class="note"></div>` instead of `[!NOTE]`.

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +312,36 @@ Please check from the device Web UI that the configured server is reachable.
+  - **Internet & Security** >> **SNTP Server** (make sure this server is reachable)
+- Consider **TEMPORARILY** enabling Debug Log in **Settings** >> **DEVICE INFO**, just long enough to capture any current errors the Shelly device is reporting, and then disabling the Debug Log.
+
+> [!WARNING]  
```
You should use `<div class="note warning"></div>`instead of `[!WARNING]`.

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +312,36 @@ Please check from the device Web UI that the configured server is reachable.
+- If Home Assistant is running in a virtual machine or as a service on an operating system other than Home Assistant OS, open `UDP` port `5683` on the host device's firewall, and/or ensure that communication from this port is forwarded to the Home Assistant service.
+- Check WiFi connection quality. In the device's web UI, go to **Settings** >> **DEVICE INFO**. Confirm the proper SSID name and signal strength; if device reports WiFi RSSI of -70 dBm or a greater negative value, signal strength may be the issue. You can test this hypothesis by temporarily relocating the affected Shelly device(s) an/or their WiFi access point(s) to be closer together and/or with ...
+  - Optimize WiFi access point placement to boost the worst affected devices' Wi-Fi signal. Minimizing metal obstructions (includig wires) in the immediate vicinity of Wi-Fi antennae can often make a meaningful difference.
+  - If your affected Shelly devices are installed in metal electrical boxes, **only if local electrical codes allow**, consider having a qualified electrician replace any unused metal electrical box knock-outs with NM (non-metallic) push-in connectors. This will allow  additional Wi-Fi signal pathways to the Shelly device.
+  - If your affected Shelly devices have a dense mass of electrical wiring surrounding them, consider having a qualified electrician clean up the wiring. Multi-conductor lever connectors, such as WAGO 221-415 *(example - not endorsement)*, can allow for more compact connections compared to traditional twist nuts.
```
I'm not sure we need this. This is HA integration documentation, not an electrical manual.

chefdeit replied:
This is an excellent point to ponder:

- On the one hand, the reality is,
  - many Shelly switches are situated in electrical boxes whose location and material are governed by local electrical codes, 
  - users do report very, very low signal strengths / poor connections, 
  - a very poor connection is something known to create problems.
  
This hurts the entire HA ecosystem by spoiling the user experience and adding support overhead - hence dampening HA's vitality and spread. Unmitigated poor connections (where it could be fixed with the right knowledge) belong in a lab. So whether a firewall, a subnet, a VLAN, or a metal electrical box are blocking the communication, the end result is the same. Hence, awareness ought to be the same. Not everyone has RF communications knowledge, and most electricians simply treat smart switches as just regular switches a customer provided. HA user awareness and directions at install time along the lines outlined in the PR can significantly affect the outcomes.
- On the other hand, the same RF propagation and signal strength issues can and do afflict not just Shelly but many other devices, such as Z-Wave, Zigbee, and a plethora of other wireless protocols. Hence:
  - While I believe firmly that this awareness and advice (expanding on basics already in the version currently live and offering additional options where, e.g. the current advice to relocate the electrical box with a Shelly device may be quite impractical for many installations),
  - I do agree with your point that this RF knowledge and solution ideas belongs somewhere other than Shelly integration documentation.

I'll therefore trim this section to a bare minimum and will look for a place to open a separate PR that will add this knowledge that can be linked from any integrations relying on wireless communication.

chefdeit committed:
Update shelly.markdown per PR feedback

- streamline how to identify gen1 vs gen2 shelly devices
- switch note and warning box syntax from GitHub markdown to styled divs
- make troubleshooting section at the bottom more succinct

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+There are two generations of Shelly devices, and this integration supports both. In Home Assistant UI, on each Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`. There are differences in how gen1 vs. gen2 devices should be configured, and in the naming of entities and devices.
 
-## Shelly device configuration (generation 1)
+<div class="note">Shelly cloud connection is not required for this HA integration to work. Shelly devices are communicating directly with this HA integration.</div>
```
```suggestion
<div class="note">Shelly cloud connection is not required for this integration to work. Shelly devices are communicating directly with Home Assistant server.</div>
```

We do not need to repeat that the case concerns "HA integration". Besides, I'm not a fan of abbreviations in the documentation, it's better to use the full name of Home Assistant.

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
 ## Shelly device generations
 
-There are two generations of devices. Both generations are supported by this integration. There are some differences in how devices should be configured and in the naming of entities and devices between generations.
+There are two generations of Shelly devices, and this integration supports both. In Home Assistant UI, on each Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`. There are differences in how gen1 vs. gen2 devices should be configured, and in the naming of entities and devices.
```
```suggestion
There are two generations of Shelly devices, and this integration supports both. In Home Assistant UI, on each Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`. There are differences in how generation 1 vs. generation 2 devices should be configured, and in the naming of entities and devices.
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+## Configuring 1st generaton Shelly devices
 
-Home Assistant will display a repair issue for the Shelly device if push updates from this device do not reach the Home Assistant server.
+First generation Shelly devices use the `CoIoT` protocol to communicate with the HA integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:
```
```suggestion
First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
-- If Home Assistant is running as a virtual machine or service on an operating system other than Home Assistant OS, you should open `UDP` port `5683` on the device's firewall and/or ensure that communication from this port is redirected to the Home Assistant service.
-- The missing push updates may be related to the WiFi network range. If using a WiFi network with several access points, enable **Internet & Security** >> **WiFi Client AP Roaming** option. Consider moving Shelly device closer to the WiFi access point. Consider adding another WiFi access point, which will improve the connection quality with the device.
-- If you think your Shelly devices are working correctly and don't want to change your network/configuration, you can ignore the repair issue. Still, you must know you are giving up the best experience of using first-generation Shelly devices with Home Assistant.
+If you have followed the above steps, but the HA repair issue remains, there are diagnostic steps and ideas listed in the **Known issues and limitations** section at the bottom of this document.
```
```suggestion
If you have followed the above steps, but the repair issue remains, there are diagnostic steps and ideas listed in the **Known issues and limitations** section at the bottom of this document.
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+## Configuring 2nd generation Shelly devices
 
-`ws://` + `Home_Assistant_local_ip_address:Port` + `/api/shelly/ws` (for example: `ws://192.168.1.100:8123/api/shelly/ws`), click **Apply** to save the settings.
+Second generation Shelly devices use the `RPC` protocol to communicate with the integration. _Battery powered gen. 2 devices **only**_ need manual outbound WebSocket configuration:
```
```suggestion
Second generation Shelly devices use the `RPC` protocol to communicate with the integration. Battery powered 2nd generation devices **only** need manual outbound WebSocket configuration:
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;
+2. In the device's web UI (at the local IP address of your Shelly device), go to: Left sidebar >> **Settings** >> **Outbound WebSocket**;
+3. Make sure box **Enable Outbound WebSocket** is selected;
+4. In Server field, enter ws://`Your_Home_Assistant_local_IP_address:Port`/api/shelly/ws (note: port must match your HA port when you open HA UI);
```
```suggestion
4. In Server field, enter `ws://Your_Home_Assistant_local_IP_address:Port/api/shelly/ws` (note: port must match your Home Assistant port when you open the UI);
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+2. In the device's web UI (at the local IP address of your Shelly device), go to: **Internet & Security** >> **ADVANCED - DEVELOPER SETTINGS**;
+3. Make sure box **Enable CoIoT** is selected;
+4. In **CoIoT peer** field, enter `Your_Home_Assistant_local_IP_address:5683`
+5. **Save**;
```
```suggestion
5. Click **Save**;
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +307,31 @@ Please check from the device Web UI that the configured server is reachable.
 
 ## Known issues and limitations
 
+- Generation 1 "Shelly 4Pro" and "Shelly Sense", *based on the old CoAP v1 protocol*, are **not** supported.
```
```suggestion
- Generation 1 "Shelly 4Pro" and "Shelly Sense", based on the old CoAP v1 protocol, are not supported.
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +307,31 @@ Please check from the device Web UI that the configured server is reachable.
+
+### Troubleshooting first generation Shelly devices raising a HA repair issue
+
+The following ideas might help you diagnose and fix the problem of the gen. 1 Shelly device CoIoT push updates not propagating to HA, and causing HA to display a repair issue:
```
```suggestion
The following ideas might help you diagnose and fix the problem of 1st generation Shelly device CoIoT push updates not propagating to Home Assistant:
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +307,31 @@ Please check from the device Web UI that the configured server is reachable.
+
+The following ideas might help you diagnose and fix the problem of the gen. 1 Shelly device CoIoT push updates not propagating to HA, and causing HA to display a repair issue:
+
+- Check that communication on `UDP` port `5683` is propagating from your Shelly device to Home Assistant. This can be affected when the HA is behind a firewall (such as one running on HA's virtual machine host) or in a different subnet or VLAN than Shelly devices. Connect a computer to the same WiFi SSID at the same location / access point as the affected Shelly device(s) and try:
```
```suggestion
- Check that communication on `UDP` port `5683` is propagating from your Shelly device to Home Assistant. This can be affected when the Home Assistant server is behind a firewall (such as one running on the virtual machine host) or in a different subnet or VLAN than Shelly devices. Connect a computer to the same WiFi SSID at the same location/access point as the affected Shelly device(s) and try:
```

bieniu commented on source/_integrations/shelly.markdown:
```diff
@@ -301,9 +307,31 @@ Please check from the device Web UI that the configured server is reachable.
+
+- Check that communication on `UDP` port `5683` is propagating from your Shelly device to Home Assistant. This can be affected when the HA is behind a firewall (such as one running on HA's virtual machine host) or in a different subnet or VLAN than Shelly devices. Connect a computer to the same WiFi SSID at the same location / access point as the affected Shelly device(s) and try:
+  - On Windows command line: `portqry.exe -n Your_Home_Assistant_local_IP_address -p udp -e 5683 -sp 5683` This tool is available at https://www.microsoft.com/en-us/download/details.aspx?id=17148 or its GUI version PortQryUI at https://www.microsoft.com/en-us/download/details.aspx?id=24009&wa=wsignin1.0
+  - On Linux: `nc -vz -u Your_Home_Assistant_local_IP_address 5683`
```
It seems to me that the user may not understand the output of these tools. Maybe an example is needed.

chefdeit replied:
Added in most recent commit: desired output identifiers and links to usage and other examples

chefdeit committed:
Update source/_integrations/shelly.markdown

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Agree re abbreviations

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Remove abbreviated "HA"

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Removed abbreviated HA

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Replace abbreviation "gen. 2" with "2nd generation" per  @bieniu
Limit strong emphasis to "only"

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Add action as it related to "Save" button

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

remove highlightig

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Spell out HA as Home Assistant per @bieniu advice

Co-authored-by: Maciej Bieniek <bieniu@users.noreply.github.com>

chefdeit committed:
Update shelly.markdown with PortQuery and NetCat command examples

Add output examples and links to more info for both tools

bieniu commented:
@Chef-de-IT There is a conflict that need to be resolved before merging. You should rebase your branch.

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+## Configuring 1st generaton Shelly devices
 
-Home Assistant will display a repair issue for the Shelly device if push updates from this device do not reach the Home Assistant server.
+First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:
```
```suggestion
First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. Suggested steps to fix:
```

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:
 
-The list below will help you diagnose and fix the problem:
+1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;
```
We should tell the minimum version, "current" is not enough: what if the device for whatever reason is not offering a new firmware for example ?

chefdeit replied:
Good idea & intention - I'm not aware if ALL devices share the same firmware version # nomenclature. The goal of this PR is to provide actionable helpful steps to as many people with this issue as possible, as soon as possible - also heeding the Home Assistant dev docs advice to keep PRs small and fast rather than make them the be-all end-all.

chefdeit replied:
Identified and added the vsersion

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;
+2. In the device's web UI (at the local IP address of your Shelly device), go to: **Internet & Security** >> **ADVANCED - DEVELOPER SETTINGS**;
+3. Make sure box **Enable CoIoT** is selected;
+4. In **CoIoT peer** field, enter `Your_Home_Assistant_local_IP_address:5683`
```
```suggestion
4. In **CoIoT peer** field, enter `Your_Home_Assistant_local_IP_address:5683`, for example: `192.168.1.10:5683`

```

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+## Configuring 2nd generation Shelly devices
 
-`ws://` + `Home_Assistant_local_ip_address:Port` + `/api/shelly/ws` (for example: `ws://192.168.1.100:8123/api/shelly/ws`), click **Apply** to save the settings.
+Second generation Shelly devices use the `RPC` protocol to communicate with the integration. Battery powered 2nd generation devices **only** need manual outbound WebSocket configuration:
```
```suggestion
Second generation Shelly devices use the `RPC` protocol to communicate with the integration. **Only** battery powered 2nd generation devices  need manual outbound WebSocket configuration:
```

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
-`ws://` + `Home_Assistant_local_ip_address:Port` + `/api/shelly/ws` (for example: `ws://192.168.1.100:8123/api/shelly/ws`), click **Apply** to save the settings.
+Second generation Shelly devices use the `RPC` protocol to communicate with the integration. Battery powered 2nd generation devices **only** need manual outbound WebSocket configuration:
+
+1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;
```
Same as above, we should specify the minimum version of the firmware

chefdeit replied:
Please suggest the change to the text with the appropriate firmware version # (I reckon whatever was current mid-August this year that is known to have the input fields that resolve the HA issue in question)

chefdeit replied:
Identified and added the version, thanks

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;
+2. In the device's web UI (at the local IP address of your Shelly device), go to: Left sidebar >> **Settings** >> **Outbound WebSocket**;
+3. Make sure box **Enable Outbound WebSocket** is selected;
+4. In Server field, enter `ws://Your_Home_Assistant_local_IP_address:Port/api/shelly/ws` (note: port must match your Home Assistant port when you open the UI);
```
I would add an example here as well

chefdeit replied:
Earlier I was asked to DELETE a whole lot of specifics because "it was getting too long and technical". Let's try to navigate this down the middle to where it actually goes in and helps.

chemelli74 commented on source/_integrations/shelly.markdown:
```diff
@@ -43,34 +43,40 @@ Integrate [Shelly devices](https://shelly.cloud) into Home Assistant.
+2. In the device's web UI (at the local IP address of your Shelly device), go to: Left sidebar >> **Settings** >> **Outbound WebSocket**;
+3. Make sure box **Enable Outbound WebSocket** is selected;
+4. In Server field, enter `ws://Your_Home_Assistant_local_IP_address:Port/api/shelly/ws` (note: port must match your Home Assistant port when you open the UI);
+5. You *might* have to tweak the **Connection Type**, depending on your local TLS certificate situation;
```
I would add the most common one and then say "you may need to tweak...."

chefdeit replied:
I wouldn't say there's a common one - different folks run their networks differently. If somebody isn't deliberate with their network security there are only 3 options and they can try "whichever one works" but I'm loathe to put it in writing like that.

chemelli74 reviewed (changes requested):
I added some comments to my flavour

chemelli74 commented:
All in all, the whole approach seems a bit too technical for me. Not really convinced in proceeding with this PR.
I appreciate your effort, but I fear it doesn't go in the right direction. Sorry.

Simone

chefdeit committed:
Update source/_integrations/shelly.markdown grammar & verbiage

Per chemelli74 grammar & verbiage suggestions

Co-authored-by: Simone Chemelli <simone.chemelli@gmail.com>

chefdeit committed:
Update source/_integrations/shelly.markdown example IP added

Co-authored-by: Simone Chemelli <simone.chemelli@gmail.com>

chefdeit committed:
Update source/_integrations/shelly.markdown

Co-authored-by: Simone Chemelli <simone.chemelli@gmail.com>

chefdeit committed:
Update shelly.markdown TLS connectin type verbiage

chefdeit committed:
Update shelly.markdown incorporate PR 28575

chefdeit committed:
Update shelly.markdown specific firmware version per Chemelli74 request

Chef-de-IT commented:
> All in all, the whole approach seems a bit too technical for me. Not really convinced in proceeding with this PR. I appreciate your effort, but I fear it doesn't go in the right direction. Sorry.
> 
> Simone

Thank you for your contributions, Simone, that were incorporated to further improve this PR of 25 commits of a Shelly integration help doc. If you feel the PR makes the help doc less helpful to enable people having the issues e.g. https://github.com/home-assistant/core/issues/97887#issuecomment-1675293319 to actually resolve them, please suggest further changes to the PR.

If you feel future Home Assistant Shelly integration code changes will prevent the issue from happening in the first place, it'll make a perfect sense, *at that time* to remove the no longer needed technical specifics from the help doc as well. If you feel there are technical specifics that are an eyesore in the PR doc such as it is right now and the users can actually benefit not seeing them, by all means please propose another change to this PR.

Many thanks!
Alex

Chef-de-IT commented:
Closing this - after a lot of revisions following a range of feedback directions from the stakeholders.
